addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -453,3 +500,16 @@ def action_open_timesheet_view_portal(self):
             'views': [(self.env.ref('hr_timesheet.timesheet_view_form_portal_user').id, 'form')],
             'context': self._context,
         }
+
+    @api.model
+    def get_unusual_days(self, date_from, date_to=None):
+        return self.env.user.employee_id._get_unusual_days(date_from, date_to)
+
+    class AccountAnalyticLineCalendarEmployee(models.Model):
+
+        _name = 'account.analytic.line.calendar.employee'
+        _description = 'Personal Filters on Employees for the Calendar view'
+
+        user_id = fields.Many2one('res.users', required=True, default=lambda self: self.env.user, ondelete='cascade', export_string_translation=False)
+        employee_id = fields.Many2one('hr.employee', export_string_translation=False)
+        active = fields.Boolean(default=True, export_string_translation=False)
```

Comment:
can you create a new file for that new model? I would prefer having one model per file. 🙂

Reply:
That's what I was thinking too, but [the official guidelines](https://www.odoo.com/documentation/18.0/contributing/development/coding_guidelines.html#file-naming) state the following:
> Concerning models, split the business logic by sets of models belonging to a same main model. Each set lies in a given file named based on its main model.

So I don't know if we should have a separate file for a tiny model that will never grow? :man_shrugging: